frontend/src/lib/components/HedgehogBuddy/hedgehogBuddyLogic.ts

```diff
@@ -54,7 +54,11 @@ export const hedgehogBuddyLogic = kea<hedgehogBuddyLogicType>([
             {
                 loadRemoteConfig: async () => {
                     const endpoint = '/api/users/@me/hedgehog_config'
-                    if (toolbarConfigLogic?.findMounted()?.values.temporaryToken) {
+                    const maybeMountedToolbarConfigLogic = toolbarConfigLogic.findMounted()
+                    if (maybeMountedToolbarConfigLogic) {
+                        if (!maybeMountedToolbarConfigLogic.values.isAuthenticated) {
+                            return null
+                        }
                         return await (await toolbarFetch(endpoint, 'GET')).json()
                     }
                     return await api.get<Partial<HedgehogConfig>>(endpoint)
```

Comment:
i find this hard to parse...

```
 const isInTheToolbar = toolbarConfigLogic.findMounted()
                    if (isInTheToolbar) {
                        if (isInTheToolbar.values.isAuthenticated) {
                            return await (await toolbarFetch(endpoint, 'GET')).json()
                        }
                        return null
                    }
```

maybe that's slightly easier but not hugely 🤷

Reply:
I see what you mean. Although also not sure about `isInTheToolbar`, as that looks much like a bool. Hopefully this is  also better: [`775a336` (#25323)](https://github.com/PostHog/posthog/pull/25323/commits/775a3364cc08269322192678d7995e158e76517f)

Reply:
> also not sure about isInTheToolbar, as that looks much like a bool

fair! 

nice compromise :)